src/package.cc

```diff
@@ -4,71 +4,58 @@
 namespace cyclus {
 
 // unpackaged id is 1, so start the user-declared packaging id at 2
-int Package::next_id_ = 2;
+int Package::next_package_id_ = 2;
 
-template <class T>
-double Package::GetFillMass(typename T::Ptr r) {
-  if (r->quantity() < fill_min_) {
+Package::Ptr Package::Create() {
+  Ptr p(new Package());
+  return p;
+}
+
+Package::Ptr Package::Create(std::string name, double fill_min, double fill_max, std::string strategy) {
+  if (fill_min < 0 || fill_max < 0) {
+    throw ValueError("fill_min and fill_max must be non-negative");
+  }
+  else if (fill_min > fill_max) {
+    throw ValueError("fill_min must be less than or equal to fill_max");
+  }
+  Ptr p(new Package(name, fill_min, fill_max, strategy));
+  return p;
+}
+
+Package::Ptr Package::CreateUnpackaged() {
+  Ptr p(new Package(1, "unpackaged"));
```

Comment:
Safest to use `unpackaged_id_` here and maybe introduce `unpackaged_name_` as well?  That way there can always be a single place to set/change those things.

Reply:
Also - can't this just be a call to `Package::Create("unpackaged")`?

Reply:
I was getting weirdness with testing where a new context was called but the numbering wasn't reset so it was numbering unpackaged packages with ids >1

Reply:
hmmm... I'll have to think about that!  I guess our package_id will not be unique across multiple contexts but we do want `unpackaged` to be singular across those contexts.

Reply:
New idea: what if the unpackaged object is a static member of the Package class, and we return a pointer to that package when it is requested through the context:

1. Class `Package` has member `static Package::Ptr unpackaged_;`
2. At startup, we initialize this with `id_ = Package::unpackaged_id_` and `name_ = Package::unpackaged_name_`
3. If we request a package from the context with `name == Package::unpackaged_name_` or with `id == Package::unpackaged_id_` we return `Package::unpackaged_`

That is, this object is not actually stored in the context, but those methods check first to see if this is what has been  requested.

Reply:
A static member of the class that's also an instance of the class? I either don't fully understand your suggestion or don't understand how to successfully implement that

Reply:
See the PR I made into this branch